air/src/air.rs

```diff
@@ -93,25 +93,47 @@ pub trait AirBuilder: Sized {
         self.when(self.is_transition_window(size))
     }
 
+    /// Assert that the given element is zero.
+    ///
+    /// Where possible, batching multiple assert_zero calls
+    /// into a single assert_zeros call will improve performance.
     fn assert_zero<I: Into<Self::Expr>>(&mut self, x: I);
 
+    /// Assert that every element of a given array is 0.
+    ///
+    /// This should be preferred over calling `assert_zero` multiple times.
+    fn assert_zeros<const N: usize, I: Into<Self::Expr>>(&mut self, array: [I; N]) {
+        for elem in array {
+            self.assert_zero(elem);
+        }
+    }
+
+    /// Assert that a given array consists of only boolean values.
+    fn assert_bools<const N: usize, I: Into<Self::Expr>>(&mut self, array: [I; N]) {
+        let zero_array = array.map(|x| {
+            let x = x.into();
+            x.clone() * (x - Self::Expr::ONE)
```

Comment:
Does this work as well?
```suggestion
            x.into().bool_check()
```

Reply:
Yep, thanks. Was developing two things in parallel and missed some overlap.